### ISSUE TITLE
After deleting addon, and reset zoom to 100%, zoom setting is not persistent.

### ISSUE THREAD

**Issue by freedriven** [Enchufadoo/fixedzoom#14]
I've tried multiple combinations (setting zoom to 100% then deleting the addon, deleting addon and setting zoom to 100% via normal firefox settings, etc), but, no matter what I do, when I restart Firefox, zoom level starts at 115% (the value I initial set in fixedzoom).

Deleting the addon should reset to default settings.

**Comment by Enchufadoo** [comment]
The addon doesn't modify any setting, when you delete it the browser should go back to it's normal behaviour because the zoom change is not done via modifying the configuration of the browser but running a script everytime a page loads.

Please make sure that any other addon / settings you may have modified manually are the cause of this behaviour.

**Comment by freedriven** [comment]
My apologies, I used the wrong description. I have the above problem when I disable the addon. When I delete it, it does as it should.

So, disabling the addon created the above odd behavior. Deleting it, there's no issue.

Thanks.

**Comment by Enchufadoo** [comment]
Oh now I see, I'll look into it, someone already had reported this and I thought I had fixed it.

**Comment by freedriven** [comment]
Well, I'm actually going back to my original wording. I am certain the following has happened: 
With fixedzoom installed and enabled, set zoom from 115% to 100%.
disabled fixedzoomed
removed fixedzoomed
restarted firefox, and I have websites coming up at 115% zoom!!!! Ug.

**Comment by Enchufadoo** [comment]
It's the browser remembering the zoom you last used on that particular site you are referring. If you go to a site that you never visited before the zoom should be 100%.

**Comment by freedriven** [comment]
So my assumption was, that if I set the zoom to 100% in fixedzoom before I disabled and/or deleted it, then it would set all pages zoom back to 100%. Is that not true?

**Comment by Enchufadoo** [comment]
The button only sets to 100% the open tabs, it doesn't modify any of firefox saved zoom preferences.

**Comment by sleek22** [comment]
If I may throw some light on these problems...

Firefox can set zoom by site or by tab. I've found that setting by site can be problematic as Firefox is always remembering old zoom levels for sites.

If you go into about:config and toggle the following setting to "false", you won't have any remembered sites.

browser.zoom.siteSpecific

With the above set to "false", any changes in zoom you make will be tab-specific, and if you turn the enable off on Fixed Zoom, all tabs open at 100% zoom. If you prefer site-specific zoom, then leave Fixed Zoom enabled and use its advanced features on sites that you want a specific zoom for.

**Comment by Enchufadoo** [comment]
@sleek22 I think the problem is that. He tried the extension and tried to disabled/delete it but firefox still remembers the zoom you last used so he thinks the extension still has some effect over the zoom.

I will delete the restore to 100% zoom button, it's confusing, and I can't delete firefox saved zoom preferences.

**Comment by sleek22** [comment]
> I will delete the restore to 100% zoom button

No, please don't do that. I use this feature all the time.

As an aside, Zoom Page WE has a button that allows one to "Clear All Per-Site Data". Perhaps you could add a button like that.

You can see it in the picture here as the last screenshot (bottom-right): https://addons.mozilla.org/en-US/firefox/addon/zoom-page-we/

**Comment by Enchufadoo** [comment]
No deleting then   :sweat_smile: I'll just rename it.

Reading a little of Zoom Page WE code I think it does the same thing that I do, he gets all the active tabs and sets the zoom to "defaultFullLevel" that is a setting from the extension. 

I believe that you can't clear zoom preferences from the browser. You can delete cookies and sites preferences and so on but just the zoom you can't. This is what firefox allows to delete:

- browser cache
- cookies
- downloads
- history
- local storage
- plugin data
- saved form data
- saved passwords

https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/browsingData

**Comment by sleek22** [comment]
Actually, Zoom Page WE deletes all zoom preferences even in unloaded sites.

I know this because this is the add-on I was using before installing yours. Whenever I had too many remembered zoom levels, I would hit that button and get a fresh start. No custom zoom levels were remembered.

Perhaps that add-on was overriding stored zoom levels? Like in a roundabout way used a stored database or something?

**Comment by Enchufadoo** [comment]
Yes, it does have a database with urls and their corresponding zoom level, so I guess it does override stored settings.

It works pretty much like this extension, why did you stop using it if I may ask?

**Comment by sleek22** [comment]
In the Web Extension version of Zoom Page WE, unlike the previous versions, changing default zoom would only take effect on new tabs but not the currently opened tabs. So I would have to refresh all the tabs.

In Fixed Zoom, I can zoom or restore all tabs to 10% and the effect is instantaneous.

I spend about 14 hours a day on the computer so from time to time, I will use Fixed Zoom to zoom all tabs to 180% and read the computer standing up. When I sit down again, I restore the zoom level on all tabs to 100% by disabling Fixed Zoom in the icon, and restoring  the zoom to 100%.

Also, in your add-on, there are a lot less clicks. One click to enable my preset zoom of 180%. Two clicks to restore, one to disable and then pressing the "Restore zoom to 100%" link in your add-on button.

**Comment by Enchufadoo** [comment]
Oh I see! I never imagine how others use it, if I had known about zoom page we I wouldn't even have bothered to make the extension because it does all I need 😄.

Would having shortcuts to most used zooms be of use? something like bookmark a value.
